### PR TITLE
CRM-17115 extra/ stdise fields on CiviReport

### PR DESCRIPTION
**Pull request by eileenmcnaughton**
- [CRM-17115: Add contact detail & address fields to membership report](https://issues.civicrm.org/jira/browse/CRM-17115)

### CODE REVIEW

File: CRM/Report/Form/Contribute/Summary.php [hunk 1]
```diff
@@ -827,6 +811,7 @@ public function alterDisplay(&$rows) {
       }
 
       $entryFound = $this->alterDisplayAddressFields($row, $rows, $rowNum, 'contribute/detail', 'List all contribution(s) for this ') ? TRUE : $entryFound;
+      $entryFound = $this->alterDisplayContactFields($row, $rows, $rowNum, 'contribute/detail') ? TRUE : $entryFound;
```
Comment (yashodha) [inline CRM/Report/Form/Contribute/Summary.php]: There seems to be a mis-match in the arguments that function 'alterDisplayContactFields' is expecting.

Comment (eileenmcnaughton) [inline CRM/Report/Form/Contribute/Summary.php]: Thanks Yashi, I altered it so the signatures for the 2 functions match & updated the PR. I assigned the JIRA back to Jitendra to re-QA. I didn't replicate all the e-notices Jitendra hit so will await feedback on that.

File: CRM/Report/Form.php [hunk 2]
```diff
@@ -4172,6 +4173,41 @@ public function alterDisplayAddressFields(&$row, &$rows, &$rowNum, $baseUrl, $ur
...
+    }
+    if (array_key_exists('civicrm_contact_suffix_id', $row)) {
+      $suffixes = CRM_Contact_BAO_Contact::buildOptions('suffix_id');
+      if ($value = $row['civicrm_contact_suffix_id']) {
+        $rows[$rowNum]['civicrm_contact_suffix_id'] = $suffixes[$rows[$rowNum]['civicrm_contact_suffix_id']];
+      }
+      $entryFound = TRUE;
+    }
+    if (array_key_exists('civicrm_contact_gender_id', $row)) {
+      $genders = CRM_Contact_BAO_Contact::buildOptions('gender_id');
```
Comment (colemanw) [inline CRM/Report/Form.php]: I wonder if using `CRM_Core_Pseudoconstant::getLabel()` would be more concise?

Comment (eileenmcnaughton) [inline CRM/Report/Form.php]: probably - I didn't know about that when I did this. If I need to re-visit this PR I'll change it

File: CRM/Report/Form/Contribute/Summary.php (hunk 1 above)
Comment (eileenmcnaughton) [inline CRM/Report/Form/Contribute/Summary.php]: looks like I hadn't succeeded in pushing when I said that - pushed now

**Comment by eileenmcnaughton** [comment]
unrelated failure

File: CRM/Report/Form.php (hunk 2 above)
Comment (seamuslee001) [inline CRM/Report/Form.php]: @eileenmcnaughton I was just upding my own custom reports and i think the best is Change line 4206 to  ``` $rows[$rowNum]['civicrm_contact_gender'] = CRM_Core_Pseudoconstant::getLabel('CRM_Contact_BAO_Contact', 'gender_id', $value);```
Then you can remove 4204

**Comment by eileenmcnaughton** [comment]
Happy now? BTW - if you can put the changes through their paces & all looks good please comment back on the PR or the ticket (or both) since it's in QA

**Comment by eileenmcnaughton** [comment]
Merging per QA by jitendra

**Pull request merged**